modules/ROOT/pages/provisioning-aliyun.adoc

```diff
@@ -0,0 +1,185 @@
+
+== Prerequisites
+
+Before provisioning a FCOS machine, you must have an Ignition configuration file containing your customizations. If you do not have one, see xref:producing-ign.adoc[Producing an Ignition File].
+
+You also need to have access to an Alibaba Cloud account and https://www.alibabacloud.com/help/doc-detail/31884.htm?spm=a2c63.p38356.879954.10.3d1264baRYHfmB#task-njz-hf4-tdb[activated Object Storage Service (OSS)]. The examples below use the https://www.alibabacloud.com/help/product/29991.htm[Alibaba Cloud CLI].
+
+== Downloading an Alibaba Cloud image
+
+Fedora CoreOS is designed to be updated automatically, with different schedules per stream.
+Once you have picked the relevant stream, download, verify, and decompress the latest Alibaba Cloud image:
+
+[source, bash]
+----
+STREAM="stable"
+coreos-installer download --decompress -s "${STREAM}" -p aliyun -f qcow2.xz
+----
+
+Alternatively, you can manually download an Alibaba Cloud image from the https://getfedora.org/coreos/download?tab=cloud_operators[download page].  Verify the download, following the instructions on that page, and decompress it.
+
+== Uploading the image to Alibaba Cloud
+
+. Create any bucket that don't already exist in your Alibaba Cloud account with a globally unique name or reuse an existing bucket:
++
+.Example creating Alibaba Cloud OSS (Object Storage Service) bucket
+[source, bash]
+----
+REGION="eu-central-1"
+BUCKET_NAME="my-bucket"
+BUCKET_URL="oss://${BUCKET_NAME}"
+aliyun oss mb "${BUCKET_URL}" --region "${REGION}"
+----
+
+. Upload an FCOS image:
++
+.Example uploading FCOS to a Alibaba Cloud OSS bucket
+[source, bash]
+----
+DOWNLOADED_IMAGE="./image.qcow2"
+IMAGE_NAME="my-fcos-image"
+IMAGE_BLOB="${IMAGE_NAME}.qcow2"
+aliyun oss cp "${DOWNLOADED_IMAGE}" "${BUCKET_URL}/${IMAGE_BLOB}" --region "${REGION}"
+----
+
+. Import uploaded FCOS image:
++
+.Example importing FCOS to Alibaba Cloud ECS
+[source, bash]
+----
+aliyun ecs ImportImage \
+    --RegionId="${REGION}" \
+    --DiskDeviceMapping.1.OSSBucket="${BUCKET_NAME}" \
+    --DiskDeviceMapping.1.OSSObject="${IMAGE_BLOB}" \
+    --ImageName="${IMAGE_NAME}"
+# Wait until the image was successfully imported
+while ! aliyun ecs DescribeImages --ImageName="${IMAGE_NAME}" | grep -q "${IMAGE_NAME}"; do sleep 10; done
+----
+
+. Determine id of imported FCOS image:
++
+.Example determining id of the imported FCOS image
+[source, bash]
+----
+IMAGE_ID=$(aliyun ecs DescribeImages --ImageName="${IMAGE_NAME}" \
+    --output cols=ImageId rows=Images.Image | head -n3 | tail -n1)
+----
+
+. Delete uploaded blob
++
+.Example deleting uploaded blob
+[source, bash]
+----
+aliyun oss rm "${BUCKET_URL}/${IMAGE_BLOB}" --region "${REGION}"
+----
+
+== Creating a VSwitch (Optional)
+
+There exists no default VPCs or VSwitches in Alibaba Cloud. Hence for creating any instances a VSwitch has to exists. Pick some existing or create one with the following steps.
```

Comment:
I seem to remember instances can be created in a standalone way with a floating public IP directly attached to them, am I wrong?
While I acknowledge that a VPC is a better infrastructure, standalone addressable nodes are usually easier to grasp for people that are getting started. What about documenting that first, and leaving the VPC setup to a later advanced section?

(Assuming my memory is not mixing details from other clouds, in which case just disregard this comment)

Reply:
I don't know anything about something like "standalone addressable nodes". I never saw such an option in the Console or in the API documentation. Currently, only VPC networks are supported, see [https://www.alibabacloud.com/help/doc-detail/61651.htm](Network Types).